daemon/daemon_main.go

```diff
@@ -901,16 +916,61 @@ func initEnv(cmd *cobra.Command) {
 		log.WithError(err).Fatal("Invalid sidecar-istio-proxy-image regular expression")
 		return
 	}
+
+	if option.Config.PolicyEnforcementInterface == "" {
+		option.Config.PolicyEnforcementInterface = viper.GetString("policy-enforcement-interface")
+	}
+
+	if !option.Config.PolicyEnforcementCleanUp {
+		option.Config.PolicyEnforcementCleanUp = viper.GetBool("policy-enforcement-clean-up")
+	}
+}
+
+// setHostDeviceWhenReady sets the given ifaceName as the HostDevice. If
+// ifaceName is not found, then it will wait forever until the device is
+// created before it assign it to the hostDevice.
+func setHostDeviceWhenReady(ifaceName string) {
+	for i := 0; ; i++ {
+		if i%10 == 0 {
+			log.WithField("interface", ifaceName).
+				Info("Waiting for the underlying interface to be initialized with containers")
+		}
+		_, err := netlink.LinkByName(ifaceName)
+		if err == nil {
+			log.WithField("interface", ifaceName).
+				Info("Underlying interface initialized with containers!")
+			break
+		}
+		time.Sleep(time.Second)
+	}
+	defaults.HostDevice = ifaceName
```

Comment:
Modifying a value in the defaults package at runtime is very surprising. Shouldn't mutable options be held in the `option` package?